backend/src/main/java/com/redhat/cloud/notifications/auth/kessel/producer/inventory/KesselInventoryClientsProducer.java

```diff
@@ -0,0 +1,81 @@
+package com.redhat.cloud.notifications.auth.kessel.producer.inventory;
+
+import com.redhat.cloud.notifications.config.BackendConfig;
+import io.quarkus.logging.Log;
+import jakarta.annotation.Priority;
+import jakarta.enterprise.context.ApplicationScoped;
+import jakarta.enterprise.inject.Alternative;
+import jakarta.enterprise.inject.Disposes;
+import jakarta.enterprise.inject.Produces;
+import jakarta.inject.Inject;
+import org.eclipse.microprofile.config.inject.ConfigProperty;
+import org.project_kessel.client.InventoryGrpcClientsManager;
+import org.project_kessel.client.NotificationsIntegrationClient;
+
+/**
+ * We have implemented our own producer for the clients because we need our
+ * clients to be "@ApplicationScoped" so that we can mock them in our tests.
+ * The default producer from the library gives us non-mockable and dependent
+ * beans, which we cannot use in our tests.
+ */
+@ApplicationScoped
+@Priority(1)
```

Comment:
Yes, if I don't add the priority the `@Inject` annotations seem to inject the beans from the libraries instead of our providers, and since we don't have the libraries' configuration properties set up, the tests break.